packages/app-store/googlecalendar/api/callback.ts

```diff
@@ -50,6 +50,34 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     },
   });
 
+  // Fetch the user's email from the Google OAuth2 userinfo endpoint using the access token in 'key'.
+  const userEmail = await fetch("https://www.googleapis.com/oauth2/v3/userinfo", {
+    method: "GET",
+    headers: {
+      Authorization: `Bearer ${key?.access_token}`,
+    },
+  })
+    .then((res) => res.json())
+    .then((res) => res.email); // Extract the user's email from the response.
+
+  // Upsert (Insert or Update) a record of primary calendar in the 'selectedCalendar' table in the Prisma database.
+  await prisma.selectedCalendar.upsert({
+    where: {
+      userId_integration_externalId: {
+        userId: req.session.user.id,
+        integration: credentials?.type, // Match the credential type.
+        externalId: userEmail, // Match the user's email.
```

Comment:
`question`: Would externalId of `primary` do the same as providing the user's email?

Reply:
@vikaspatil0021 I'm going to back up @exception on this one. Not only would that reduce one more call we have to make to Google but that means we don't need to add any new scope to our OAuth request. If we add to the scope then any current access tokens won't be able to access the new scope and could cause some unintended side effects.

Reply:
> `question`: Would externalId of `primary` do the same as providing the user's email?

Yes, 
For Google's primary calendar, the user's email is used as the external ID.
TRPC output for primary calendar
![Screenshot 2023-10-18 154241](https://github.com/calcom/cal.com/assets/112180774/a6a3957a-d264-4bf5-b5f4-21715969f830)
for other calendars
![Screenshot 2023-10-18 154509](https://github.com/calcom/cal.com/assets/112180774/3b5864d3-58e3-4e06-b68f-f96daa03a9a3)

Reply:
> @vikaspatil0021 I'm going to back up @exception on this one. Not only would that reduce one more call we have to make to Google but that means we don't need to add any new scope to our OAuth request. If we add to the scope then any current access tokens won't be able to access the new scope and could cause some unintended side effects.

@joeauyeung Thank you for your feedback and support

About scopes, any modifications made in the "add.ts" and "callback.ts" files should not trigger for existing access tokens. These files run only during the initial installation of the Google Calendar integration.
Please correct me if I'm mistaken.

Reply:
@joeauyeung I believe @vikaspatil0021 is correct, however, I'm not sure if upon requesting a refresh token and the scopes having changed there will be any issues. I would assume not, but better be safe than sorry.